base/nullable.jl

```diff
@@ -78,3 +78,61 @@ function hash(x::Nullable, h::UInt)
         return hash(x.value, h + nullablehash_seed)
     end
 end
+
+# as collection
+length(x::Nullable) = x.isnull ? 0 : 1
+endof(x::Nullable)  = length(x)
+
+# indexing is either without index, or with 1 as index
+# generalized linear indexing is not supported
+linearindexing{T}(::Nullable{T}) = LinearFast()
+function getindex(x::Nullable)
```

Comment:
Let's discuss this elsewhere if we really want to change it. This PR has enough controversies in it.